aten/src/ATen/native/Convolution.cpp

```diff
@@ -422,6 +422,9 @@ struct ConvParams {
   // cudnn and miopen are guaranteed not to be on mobile, and T102591915 / T110194934 suggest
   // that maybe the compiledWithCuDNN() check sometimes segfaults (though I can't imagine how)
 #if !defined(C10_MOBILE)
+    if (at::globalContext().userForceCuDNN()) {
```

Comment:
Does this mean that we will return True for `use_cudnn` if `force_cudnn` even if `compiledWithCuDNN` is False?

Reply:
That's a good point, originally I was thinking that `force_cudnn` should raise any possible failures, but in this case it's probably better to just return a warning about `force_cudnn` not having the intended effect.